ext/drreg/drreg.c

```diff
@@ -65,17 +65,65 @@
 
 #define PRE instrlist_meta_preinsert
 
-/* This should be pretty hard to exceed as there aren't this many GPRs */
-#define MAX_SPILLS (SPILL_SLOT_MAX + 8)
+/* drreg uses DR spill slots by default if no addressable TLS
+ * storage is available. Since these slots are shared also with clients,
+ * it could be the case that the client uses a lot of these slots, resulting
+ * in an insufficient number of available slots for drreg to function properly.
+ *
+ * In order to address this concern, we use a hard-coded max on the number
+ * of spill slots tracked. This number of spills intentionally surpasses
+ * SPILL_SLOT_MAX to cater both for drreg and possible clients,
+ * It is set to be pretty hard to exceed as there aren't that many GPRs.
+ */
+#define MAX_SPILLS (SPILL_SLOT_MAX + DR_NUM_GPR_REGS)
+/* We choose the number of available slots for spilling simds to match their
+ * theoretical max number for a given build.
+ *
+ * We add an additional slot for temporary storage used by drreg. For example,
+ * it is used to handle cross-app spilling. Note that this is in contrast to GPRs,
+ * which require allocated thread storage for cross-app spilling as DR slots are not
+ * guaranteed preserve stored data in such cases.
+ */
+#define MAX_SIMD_SPILLS (DR_NUM_SIMD_VECTOR_REGS + 1)
```

Comment:
The + 1 doesn't really make sense, if anything, we would need 2 * DR_NUM_SIMD_VECTOR_REGS. Might just leave it?

Reply:
So I added the +1 to store any cross app spilling for SIMD registers. But thinking about this now, we should be more flexible.  I am therefore in agreement with your following point concerning add num_simd_spill_slots.

Reply:
Concern dealt via the next review point.